src/frontend/components/StudentLiveAdvertising/StudentLiveRegistration/RegistrationForm/index.tsx

```diff
@@ -91,116 +123,106 @@ const messages = defineMessages({
 
-  let inputWidth: string;
-  let submitButtonStyle: CSSProperties;
-  if (size === 'small') {
-    inputWidth = '100%';
-    submitButtonStyle = { width: '100%', marginTop: '16px' };
-  } else {
-    inputWidth = '50%';
-    submitButtonStyle = { minWidth: '40%' };
-  }
-  const SubmitButton = () => (
-    <Button
-      type="submit"
-      primary
-      label={intl.formatMessage(messages.formSubmitTitle)}
-      style={submitButtonStyle}
-    />
+  const displayEmailInput = !(
+    isLtiToken &&
+    values.email &&
+    values.email !== ''
   );
 
   return (
     <Grommet theme={formTheme}>
       <Form
         value={values}
-        onChange={(formValues) => setValues(formValues)}
+        onChange={setValues}
         onSubmit={async ({ value }) => {
-          if (isSubmitOngoing.current) {
-            //  a submit is already in progress
+          if (!value.email) {
+            //  submit without email should not be possible since validation should fail
             return;
           }
+
+          let anonymousId: Maybe<string>;
+          if (!isLtiToken) {
+            anonymousId = getAnonymousId();
+          }
+          await createLiveRegistration(value.email, anonymousId);
+          setRegistrationCompleted();
+        }}
+        onSubmitError={(value, error) => {
           if (!value.email) {
             //  submit without email should not be possible since validation should fail
-            return;
+            return {};
+          }
+
+          let errorMessage;
+          if (error.email && error.email[0].indexOf('already registered') > 0) {
+            errorMessage = intl.formatMessage(
+              messages.updateMailAlreadyExistingError,
+              value,
+            );
+          } else if (error.email) {
+            errorMessage = intl.formatMessage(
+              messages.updateMailNotValidError,
+              value,
+            );
+          } else {
+            errorMessage = intl.formatMessage(
+              messages.updateMailDefaultError,
+              value,
+            );
           }
 
-          isSubmitOngoing.current = true;
-          await onSubmit(value.email);
-          isSubmitOngoing.current = false;
+          return { email: errorMessage };
         }}
       >
-        <FormField
-          name="email"
-          htmlFor="text-input-id"
-          contentProps={{ pad: 'none', style: { border: 'none' } }}
-          pad={false}
-          validate={{
-            regexp:
-              /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/,
-            message: intl.formatMessage(messages.emailFormatValidationError),
-            status: 'error',
-          }}
-        >
-          <RegistrationEmailField
-            inputLabel={intl.formatMessage(messages.emailInputLabel)}
-            id="text-input-id"
-            inputWidth={inputWidth}
+        {displayEmailInput && (
```

Comment:
What if value set in the hidden fiel raise an error server side ?

Reply:
You mean if the mail is invalid or already registered ?

Reply:
For some reason, if the mail in the lti request fail to validate in the backend (or frontend) but the user is not yet registered

Reply:
A generic error message should be display then. Something like try again later.

Reply:
I did this
<img width="686" alt="Capture d’écran 2022-03-01 à 19 06 23" src="https://user-images.githubusercontent.com/10722639/156226511-29d6eb61-8f6d-42cf-a0b6-0a8935eb6bce.png">